tests/templates/test_embeddings/test_angle.py

```diff
@@ -0,0 +1,304 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the AngleEmbedding template.
+"""
+import pytest
+import numpy as np
+from pennylane import numpy as pnp
+import pennylane as qml
+
+
+class TestDecomposition:
+    """Tests that the template defines the correct decomposition."""
+
+    @pytest.mark.parametrize("features", [[1, 1, 1], [1, 1]])
+    def test_expansion(self, features):
+        """Checks the queue for the default settings."""
+
+        op = qml.templates.AngleEmbedding(features=features, wires=range(4))
+        tape = op.expand()
+
+        assert len(tape.operations) == len(features)
+        for gate in tape.operations:
+            assert gate.name == "RX"
+
+    def test_state_rotx(self, ):
+        """Checks the state produced using the rotation='X' strategy."""
+
+        features = [np.pi / 2, np.pi / 2, np.pi / 4, 0]
+        dev = qml.device("default.qubit", wires=4)
+
+        @qml.qnode(dev)
+        def circuit(x=None):
+            qml.templates.AngleEmbedding(features=x, wires=range(4), rotation="X")
+            qml.PauliX(wires=0)
+            qml.templates.AngleEmbedding(features=x, wires=range(4), rotation="X")
+            return [qml.expval(qml.PauliZ(i)) for i in range(4)]
+
+        res = circuit(x=features)
+        target = [1, -1, 0, 1]
+
+        assert np.allclose(res, target)
+
+    def test_state_roty(self):
+        """Checks the state produced using the rotation='Y' strategy."""
+
+        features = [np.pi / 2, np.pi / 2, np.pi / 4, 0]
+        dev = qml.device("default.qubit", wires=4)
+
+        @qml.qnode(dev)
+        def circuit(x=None):
+            qml.templates.AngleEmbedding(features=x, wires=range(4), rotation="Y")
+            qml.PauliX(wires=0)
+            qml.templates.AngleEmbedding(features=x, wires=range(4), rotation="Y")
```

Comment:
I don't remember why we used this circuit with two templates, but I didn't have time to change...